code/src/components/app-snackbar/AppSnackbar.test.tsx

```diff
@@ -0,0 +1,38 @@
+import { screen, fireEvent } from "@testing-library/react";
+import AppSnackbar from "@/components/app-snackbar/AppSnackbar";
+import { renderWithProviders } from "@/utils/test-utils";
+import useSnackbar from "@/hooks/use-snackbar/useSnackbar";
+
+jest.mock("@/hooks/use-snackbar/useSnackbar", () => ({
+  __esModule: true,
+  default: jest.fn()
+}));
+
+const mockCloseSnackbar = jest.fn();
+
+const mockUseSnackbar = {
+  isOpen: true,
+  config: { message: "Test message", variant: "success" },
+  closeSnackbar: mockCloseSnackbar
+};
+
+describe("AppSnackbar", () => {
+  test("renders correctly", () => {
+    (useSnackbar as jest.Mock).mockReturnValueOnce(mockUseSnackbar);
```

Comment:
we have repeated this logic for tests with `mockReturnValueOnce` and also `renderWithProviders` maybe move it in `beforeEach` block ?

Reply:
Thanks for useful comment. It was worth refactoring, which I have done.